[BUG][Java] ApiClient.java: error: no suitable method found for create(byte[],MediaType)

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [] What's the actual output vs expected output?
- [x] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description
Using the java generator against a basic spec, the generated project fails to compile.
<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->

##### openapi-generator version
5.1.0 and latest master (docker image 3c6b0382cd49)

Regression from 5.0.1

<!-- which version of openapi-generator are you using, is it a regression? -->

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  version: 1.0.0
  title: Api

paths:
  /version:
    get:
      tags:
      - version
      summary: Gets the system version check
      operationId: getVersion
      responses:
        200:
          description: Success
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/VersionGetResponse'

components:
  schemas:
    VersionGetResponse:
      type: string
      description: The version of this service
      required:
        - status
      properties:
        status:
          type: string
          description: OK

```

##### Generation Details

<!-- 
    Prefer CLI steps, including the language, libraries and various options. 
    Providing config-based settings allows for simpler testing across CLI and plugins. 
    For examples, see https://github.com/OpenAPITools/openapi-generator/tree/master/bin/configs
-->

`java -jar openapi-generator-cli-5.1.0.jar generate -g java -i service.yaml`

##### Steps to reproduce

1. Run the generator as described above
2. Try to compile the code with `./gradlew compileJava` or similar

###### Expected:
It will compile

###### Actual:
A bunch of:
```
out\java\src\main\java\org\openapitools\client\ApiClient.java:818: error: no suitable method found for create(byte[],MediaType)
            return RequestBody.create((byte[]) obj, MediaType.parse(contentType));
                              ^
    method RequestBody.create(MediaType,String) is not applicable
      (argument mismatch; byte[] cannot be converted to MediaType)
    method RequestBody.create(MediaType,ByteString) is not applicable
      (argument mismatch; byte[] cannot be converted to MediaType)
    method RequestBody.create(MediaType,byte[]) is not applicable
      (argument mismatch; byte[] cannot be converted to MediaType)
    method RequestBody.create(MediaType,File) is not applicable
      (argument mismatch; byte[] cannot be converted to MediaType)
```
<!-- unambiguous set of steps to reproduce the bug.-->

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

> Should it be type: object instead?

I don't believe so - I'll look at the response, it's intended to be a simple string.

> I did a test with the latest master and the output compiles without issues

You used the attached YAML and it compiles? On both this sample and the real YAML I distilled it from it continues to not compile for me with the Docker image 563bdd246a47.


```
docker run --rm -v "${PWD}:/local" openapitools/openapi-generator-cli generate -i /local/service.yaml -g java -o /local
<...snip generation output...>

> Task :compileJava FAILED
C:\Users\tobmur\git\myproject\src\main\yaml\out\java2\src\main\java\org\openapitools\client\ApiClient.java:818: error: no suitable method found for create(byte[],MediaType)
            return RequestBody.create((byte[]) obj, MediaType.parse(contentType));
                              ^
    method RequestBody.create(MediaType,String) is not applicable
      (argument mismatch; byte[] cannot be converted to MediaType)
    method RequestBody.create(MediaType,ByteString) is not applicable
      (argument mismatch; byte[] cannot be converted to MediaType)
    method RequestBody.create(MediaType,byte[]) is not applicable
      (argument mismatch; byte[] cannot be converted to MediaType)
    method RequestBody.create(MediaType,File) is not applicable
      (argument mismatch; byte[] cannot be converted to MediaType)
C:\Users\tobmur\git\myproject\src\main\yaml\out\java2\src\main\java\org\openapitools\client\ApiClient.java:821: error: no suitable method found for create(File,MediaType)
            return RequestBody.create((File) obj, MediaType.parse(contentType));
                              ^
    method RequestBody.create(MediaType,String) is not applicable
      (argument mismatch; File cannot be converted to MediaType)
    method RequestBody.create(MediaType,ByteString) is not applicable
      (argument mismatch; File cannot be converted to MediaType)
    method RequestBody.create(MediaType,byte[]) is not applicable
      (argument mismatch; File cannot be converted to MediaType)
    method RequestBody.create(MediaType,File) is not applicable
      (argument mismatch; File cannot be converted to MediaType)
C:\Users\tobmur\git\myproject\src\main\yaml\out\java2\src\main\java\org\openapitools\client\ApiClient.java:829: error: no suitable method found for create(String,MediaType)
            return RequestBody.create(content, MediaType.parse(contentType));
                              ^
    method RequestBody.create(MediaType,String) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,ByteString) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,byte[]) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,File) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
C:\Users\tobmur\git\myproject\src\main\yaml\out\java2\src\main\java\org\openapitools\client\ApiClient.java:1079: error: no suitable method found for create(String,MediaType)
                reqBody = RequestBody.create("", MediaType.parse(contentType));
                                     ^
    method RequestBody.create(MediaType,String) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,ByteString) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,byte[]) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,File) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
C:\Users\tobmur\git\myproject\src\main\yaml\out\java2\src\main\java\org\openapitools\client\ApiClient.java:1230: error: no suitable method found for create(File,MediaType)
                mpBuilder.addPart(partHeaders, RequestBody.create(file, mediaType));
                                                          ^
    method RequestBody.create(MediaType,String) is not applicable
      (argument mismatch; File cannot be converted to MediaType)
    method RequestBody.create(MediaType,ByteString) is not applicable
      (argument mismatch; File cannot be converted to MediaType)
    method RequestBody.create(MediaType,byte[]) is not applicable
      (argument mismatch; File cannot be converted to MediaType)
    method RequestBody.create(MediaType,File) is not applicable
      (argument mismatch; File cannot be converted to MediaType)
C:\Users\tobmur\git\myproject\src\main\yaml\out\java2\src\main\java\org\openapitools\client\ApiClient.java:1233: error: no suitable method found for create(String,<null>)
                mpBuilder.addPart(partHeaders, RequestBody.create(parameterToString(param.getValue()), null));
                                                          ^
    method RequestBody.create(MediaType,String) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,ByteString) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,byte[]) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
    method RequestBody.create(MediaType,File) is not applicable
      (argument mismatch; String cannot be converted to MediaType)
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
6 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1s
1 actionable task: 1 executed
```
I'm on Windows with this, I wouldn't imagine this is a platform difference but maybe?

--- a/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/build.gradle.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/build.gradle.mustache
@@ -104,8 +104,8 @@ if(hasProperty('target') && target == 'android') {
 dependencies {
     implementation 'io.swagger:swagger-annotations:1.5.24'
     implementation "com.google.code.findbugs:jsr305:3.0.2"
-    implementation 'com.squareup.okhttp3:okhttp:3.14.7'
-    implementation 'com.squareup.okhttp3:logging-interceptor:3.14.7'
+    implementation 'com.squareup.okhttp3:okhttp:4.9.1'
+    implementation 'com.squareup.okhttp3:logging-interceptor:4.9.1'
     implementation 'com.google.code.gson:gson:2.8.6'
     implementation 'io.gsonfire:gson-fire:1.8.4'
     {{#hasOAuthMethods}}

--- a/samples/client/petstore/java/okhttp-gson-dynamicOperations/build.gradle
+++ b/samples/client/petstore/java/okhttp-gson-dynamicOperations/build.gradle
@@ -100,8 +100,8 @@ if(hasProperty('target') && target == 'android') {
 dependencies {
     implementation 'io.swagger:swagger-annotations:1.5.24'
     implementation "com.google.code.findbugs:jsr305:3.0.2"
-    implementation 'com.squareup.okhttp3:okhttp:3.14.7'
-    implementation 'com.squareup.okhttp3:logging-interceptor:3.14.7'
+    implementation 'com.squareup.okhttp3:okhttp:4.9.1'
+    implementation 'com.squareup.okhttp3:logging-interceptor:4.9.1'
     implementation 'com.google.code.gson:gson:2.8.6'
     implementation 'io.gsonfire:gson-fire:1.8.4'
     implementation group: 'org.apache.oltu.oauth2', name: 'org.apache.oltu.oauth2.client', version: '1.0.1'

--- a/samples/client/petstore/java/okhttp-gson-parcelableModel/build.gradle
+++ b/samples/client/petstore/java/okhttp-gson-parcelableModel/build.gradle
@@ -100,8 +100,8 @@ if(hasProperty('target') && target == 'android') {
 dependencies {
     implementation 'io.swagger:swagger-annotations:1.5.24'
     implementation "com.google.code.findbugs:jsr305:3.0.2"
-    implementation 'com.squareup.okhttp3:okhttp:3.14.7'
-    implementation 'com.squareup.okhttp3:logging-interceptor:3.14.7'
+    implementation 'com.squareup.okhttp3:okhttp:4.9.1'
+    implementation 'com.squareup.okhttp3:logging-interceptor:4.9.1'
     implementation 'com.google.code.gson:gson:2.8.6'
     implementation 'io.gsonfire:gson-fire:1.8.4'
     implementation group: 'org.apache.oltu.oauth2', name: 'org.apache.oltu.oauth2.client', version: '1.0.1'

--- a/samples/client/petstore/java/okhttp-gson/build.gradle
+++ b/samples/client/petstore/java/okhttp-gson/build.gradle
@@ -100,8 +100,8 @@ if(hasProperty('target') && target == 'android') {
 dependencies {
     implementation 'io.swagger:swagger-annotations:1.5.24'
     implementation "com.google.code.findbugs:jsr305:3.0.2"
-    implementation 'com.squareup.okhttp3:okhttp:3.14.7'
-    implementation 'com.squareup.okhttp3:logging-interceptor:3.14.7'
+    implementation 'com.squareup.okhttp3:okhttp:4.9.1'
+    implementation 'com.squareup.okhttp3:logging-interceptor:4.9.1'
     implementation 'com.google.code.gson:gson:2.8.6'
     implementation 'io.gsonfire:gson-fire:1.8.4'
     implementation group: 'org.apache.oltu.oauth2', name: 'org.apache.oltu.oauth2.client', version: '1.0.1'

